src/IO/S3Common.cpp

```diff
@@ -839,6 +841,26 @@ namespace S3
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "Bucket name length is out of bounds in virtual hosted style S3 URI:     {}{}",
                             quoteString(bucket), !uri.empty() ? " (" + uri.toString() + ")" : "");
     }
+
+    size_t getObjectSize(std::shared_ptr<Aws::S3::S3Client> client_ptr, const String & bucket, const String & key, bool throw_on_error)
```

Comment:
There is also `ReadBufferFromS3::getTotalSize()` with similar code, so may be delete the code there and reuse this `getObjectSize`